src/plugins/data/common/search/search_source/create_search_source.ts

```diff
@@ -55,8 +66,11 @@ export const createSearchSource = (
     return fields;
   };
 
-  const createSearchSourceFn = async (searchSourceFields: SerializedSearchSourceFields = {}) => {
-    const fields = await createFields(searchSourceFields);
+  const createSearchSourceFn = async (
+    searchSourceFields: SerializedSearchSourceFields = {},
+    useDataViewLazy: boolean
```

Comment:
1) we need to handle the hydration of searchSourceFields index differently for alerting, if we don't do this, e.g. in Discover and there are just the fields loaded, that are need, those are missing when a saved search is being loaded. We can work around of this, but I'd say, focus on the alerting relevant part

Reply:
The alternative is to create and make available both DataView and DataViewLazy objects

Reply:
yes, that's what I was thinking. That's why I was thinking, focusing on solving one problem, the more important one (alerting), keep the current way it works for the other one, so there are no tears